### PR TITLE
Numbers should be allowed to contain underscores.

### PR DESCRIPTION
**Pull request by mrmonroe**
# Description
- [x] Source branch in your fork has meaningful name (not `master`)

Fixes Issue: 
#1836 Numbers should be allowed to contain underscores.

# Before Merge Checklist 
These items can be completed after PR is created.

(Check any items that are not applicable (NA) for this PR)

- [x] JavaScript implementation
- [x] Python implementation (NA if HTML beautifier)
- [x] Added Tests to data file(s)
- [NA] Added command-line option(s) (NA if
- [NA] README.md documents new feature/option(s)

### CODE REVIEW

**Comment by mrmonroe** [comment]
Hey guys, 
Looks like the code climate failure is a long-standing issue as the regex line is 104 characters long. I'll be glad to break it down but not sure if you guys have been overriding it and don't want to break style or readability for the code. 

The other failures are from the ./tools/git-status-clear.sh script running in the Azure pipeline which looks like an outstanding issue, although I don't see it logged. I ran the script on my Mac and Linux boxes and have no problems with it. I'd be glad to help you guys with it but of course, I don't have access to the Azure account. 

Let me know how you want to proceed.

**Comment by bitwiseman** [comment]
> Looks like the code climate failure is a long-standing issue as the regex line is 104 characters long. I'll be glad to break it down but not sure if you guys have been overriding it and don't want to break style or readability for the code. 

Please feel free to break the regex to multiple lines/variables for clarity. 

Thanks for the heads up on the CI failure, I'll look into that.

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Please add tests as noted. 

Also, the current change only addresses `0b` numbers.  Will you be updating to handle all number forms?  If so, please include tests for similar scenarios to the ones noted for other numbers. 

Keep in mind `_` is the start of a variable name, so `._001` cannot be treated as a number during tokenizing.  It will need to be treated as `.` and a variable `_001`.  

There's a bunch of edge cases.   

Thanks for contributing this PR!

**Comment by mrmonroe** [comment]
@bitwiseman been down with Covid for a long time and just getting my life back together. Will get to this PR in a few days.

**Comment by bitwiseman** [comment]
@mrmonroe  
Sorry to hear that, but welcome back.  Thanks for picking this back up. Take your time.

File: test/data/javascript/tests.js [hunk 1]
```diff
@@ -4041,6 +4078,17 @@ exports.test_data = {
...
         { input: 'a=0B090x0', output: 'a = 0B0 90 x0' },
         { input: 'a=0Bg0b0o0', output: 'a = 0B g0b0o0' },
+        {
+          comment: 'Binary literals with underscore',
+          unchanged: 'a = 0b0_10011'
+        },
+        { unchanged: 'a = 0b0__10011' },
+        { unchanged: 'a = 0b_010011' },
+        { unchanged: 'a = 0b__' },
+        { unchanged: 'a = 0b0_1_1_1'},
```
Comment (bitwiseman) [inline test/data/javascript/tests.js:4088]:
```suggestion
        { unchanged: 'a = 0b0_1_1_1' },
```

File: python/jsbeautifier/javascript/tokenizer.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def __init__(self):
 dot_pattern = re.compile(r"[^\d\.]")
 
 number_pattern = re.compile(
-    r"0[xX][0123456789abcdefABCDEF]*|0[oO][01234567]*|0[bB][01]*|\d+n|(?:\.\d+|\d+\.?\d*)(?:[eE][+-]?\d+)?"
+    r"0[xX][0123456789abcdefABCDEF]*|0[oO][01234567]*|0[bB][01_]*|\d+n|(?:\.\d+|\d+\.?\d*)(?:[eE][+-]?\d+)?"
```
Comment (bitwiseman) [inline python/jsbeautifier/javascript/tokenizer.py:65]:
```suggestion
    r"0[xX][0123456789abcdefABCDEF_]*|0[oO][01234567_]*|0[bB][01_]*|\d[\d_]*n|(?:\.\d[\d_]*|\d[\d_]*\.?[\d_]*)(?:[eE][+-]?[\d_]+)?"
```

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**